### PR TITLE
Adding automatic merge job

### PR DESCRIPTION
**Pull request by adammhaile**
@tbielawa

### CODE REVIEW

File: jobs/build/merge_ocp/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                daysToKeepStr: '',
+                numToKeepStr: '1000')),
+        [
+            $class: 'ParametersDefinitionProperty',
+            parameterDefinitions: [
+                [
+                    name: 'VERSIONS',
+                    description: 'Version string for build without leading "v" (i.e. 4.0.0)',
+                    $class: 'hudson.model.StringParameterDefinition',
+                    defaultValue: "3.9,3.10,3.11,master"
```
Comment (tbielawa) [inline jobs/build/merge_ocp/Jenkinsfile]: Just curious why `master` is listed, but `4.0.0` is in the example. Is this supposed to to be multiple choice, or is the `defaultValue` just there to give you something to erase as an example?

Comment (tbielawa) [reply jobs/build/merge_ocp/Jenkinsfile]: Oh, `MERGE_VERSIONS` looks like `VERSIONS` is a comma separated input. So, maybe mention that in the `description` if that's correct?

File: jobs/build/merge_ocp/merge_ocp.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# usage: ./merge_ocp.sh ./working/ enterprise-3.11 release-3.11
+set -o xtrace
```
Comment (tbielawa) [inline jobs/build/merge_ocp/merge_ocp.sh]:
```suggestion
set -e
set -x
```

`-e` for exit on error and `-o` is just another way to print out commands before executing (like how you did `xtrace` before).

Comment (tbielawa) [reply jobs/build/merge_ocp/merge_ocp.sh]: Also probably enable the one for exiting on undefined variables.

Comment (richm) [reply jobs/build/merge_ocp/merge_ocp.sh]: `set -euxo pipefail` is sort of like `gcc -Wall` . . .

File: jobs/build/merge_ocp/Jenkinsfile (hunk 1 above)
Comment (adammhaile) [reply jobs/build/merge_ocp/Jenkinsfile]: Oops. Copy error from the job I based this on. 
Fixed and removed master from the list in lieu of checking if a given version is the current master and then handling it specially.

File: jobs/build/merge_ocp/merge_ocp.sh (hunk 2 above)
Comment (adammhaile) [reply jobs/build/merge_ocp/merge_ocp.sh]: Ack. Will change.

**Pull request merged**